Extension/src/LanguageServer/extension.ts

```diff
@@ -194,6 +218,141 @@ function onDidChangeVisibleTextEditors(editors: vscode.TextEditor[]): void {
+                let redirectUrl: string | string[];
+                if (typeof response.headers.location === "string") {
+                    redirectUrl = response.headers.location;
+                } else {
+                    redirectUrl = response.headers.location[0];
+                }
+                return resolve(downloadFileToDestination(redirectUrl, destinationPath));
+            }
+            if (response.statusCode !== 200) { // If request is not successful
+                return reject();
+            }
+            // Write file using downloaded data
+            let downloadedBytes = 0; // tslint:disable-line
+            let createdFile: fs.WriteStream = fs.createWriteStream(destinationPath);
+            response.on('data', (data) => { downloadedBytes += data.length; });
+            response.on('end', () => { createdFile.close(); });
+            createdFile.on('close', () => { resolve(); this.updateSettingsAsync(); });
+            response.on('error', (error) => { reject(); });
+            response.pipe(createdFile, { end: false });
+        });
+        request.on('error', (error) => { reject(); });
+        request.end();
+    });
+}
+
+// TODO factor this out?
+async function parseJsonAtPath(path: string): Promise<any> {
+    try {
+        const exists: boolean = await util.checkFileExists(path);
+        if (exists) {
+            const fileContent: string = await util.readFileText(path);
+            return JSON.parse(fileContent);
+        }
+    } catch (error) {
+        console.log(error);
+    }
+}
+
+async function checkAndApplyUpdate(): Promise<void> {
+    // Get current version name, stripping out everything past (and incl.) the dash if present
+    // Return if the user is using a non-Release or non-Insider build (e.g. build ending in "-master")
+    const version: string = function(): string {
+        let version: string = util.packageJson["version"];
+        const dashOffset: number = version.indexOf("-");
+        if (dashOffset !== -1) {
+            if (version.substr(dashOffset + 1) !== "insiders") {
+                return null;
+            }
+            version = version.substr(0, dashOffset); // Strip out the suffix
+        }
+        return version;
+    }();
+    if (!version) {
+        return;
+    }
+
+    // Download and parse the JSON release list from GitHub to get the latest build
+    const releaseJsonPath: string = util.getExtensionFilePath("releases.json");
+    await downloadFileToDestination("https://api.github.com/repos/Microsoft/vscode-cpptools/releases",
+                                    releaseJsonPath, { "User-Agent": "vscode-cpptools" });
+    const latestBuild: any = await parseJsonAtPath(releaseJsonPath)[0]; // [0] to get latest build
+    if (!latestBuild) {
+        return;
+    }
+
+    // Check whether the user actually needs to update
+    if (version >= latestBuild["name"]) { // latestBuild["name"] excludes version suffix
```

Comment:
I thought the latest build could still have the "-insiders" in it, no?

Reply:
This comparison is dangerous anyway since "2" > "10".  A "version" class that parses the triplet and compares the pieces numerically is safer.

Reply:
There's a "prerelease" key:bool pair in the json which determines whether it is an insiders build. It looks as though the "name" key value pair is the numbered portion only.
https://api.github.com/repos/Microsoft/vscode-cpptools/releases

Good catch on the comparison. Since the comparison is more complex, this strengthens the argument that a `version` class is the way to go

Reply:
I see `"name": "0.17.8-insiders"` in the result a few entries down.  If an insiders release is the most recent release, then the first one in the list will have `-insiders` in the name.  Am I misunderstanding?

Reply:
No, that's my bad. I had misunderstood the `prerelease` meaning. I'll fix this up; Thanks :smile: